### ISSUE TITLE
Use of mcrypt

### ISSUE THREAD

**Issue by JerWegener** [ci-bonfire/Bonfire#1275]
Function mcrypt_create_iv() is deprecated in PHP 7 and will be removed in PHP 7.2
Cheers!

**Comment by kalyan2k** [comment]
sorry what is the best way to replace mcrypt_create_iv() function?

**Comment by silverark** [comment]
Where is currently being used in Bonfire? CI has it as a fallback.

**Comment by kalyan2k** [comment]
thanks :)

**Comment by Altroo** [comment]
yeah i'm having that : Message: Function mcrypt_create_iv() is deprecated
when i activate french translation.

**Comment by Reconix** [comment]
What version of Bonfire are you using?

I enabled the French translation and didn't see that.

**Comment by Altroo** [comment]
Hello i'm using the last version & here's the error :

![sans titre](https://user-images.githubusercontent.com/30314254/31285107-630d6296-aab2-11e7-87e4-0331de68ebf2.jpg)

**Comment by Reconix** [comment]
Sorry, couldn't reproduce this error but the only other thing I can think of is do you have openSSL enabled ?

**Comment by Reconix** [comment]
Also on a side note, what browser are you using? That icon in the top right is really bothering me.

**Comment by Altroo** [comment]
I'm using google chrome & how can i check if openSSL is enabled?

**Comment by Reconix** [comment]
phpinfo() ;)

I've just checked on chrome and that icon doesn't look like that, you're testing on your local system, right ?

**Comment by Reconix** [comment]
in php.ini 
extension=php_openssl.dll
should be there to be more specific.

**Comment by Altroo** [comment]
I don't know what icon you are talking about & yeah i'm testing on xampp, & in php.ini that wasn't enabled so i enabled it but didn't work.
actually i see that error everywhere not just when using french.

![sans titre](https://user-images.githubusercontent.com/30314254/31288539-39b96db2-aabd-11e7-968f-0fed15a18605.jpg)

**Comment by Altroo** [comment]
I think the project is broken or something now when i'm logged out i can't log in lol.
Never mind that but thanks for your help anyway.

**Comment by Reconix** [comment]
Downloading xampp now to see if I can reproduce

**Comment by Reconix** [comment]
Ok so I just downloaded the latest version of Bonfire and the latest version of XAMPP and all looks good without any problems.

**Comment by Reconix** [comment]
Double checked php.ini and openSSL is indeed on by default.

So I'm not sure what else to suggest other than trying to install bonfire and xampp again.
Let me know how you can get on.

**Comment by locvfx** [comment]
I confirm you need 'https' to get rid of this error. I'm using PHP 7.2.3

**Comment by carcagi** [comment]
Hi, I am having the same issue, you can access it here (https://new.quikdrawers.com/public/). The server has openssl enabled and even has the certificate in place. Any ideas on how to solve this?

**Comment by silverark** [comment]
@carcagi I can't see the error on that page. Have you fixed/solved it since?

**Comment by truelineinfotech** [comment]
@Altroo @kalyan2k  it is completability issue with php 7.0 
## mcrypt_create_iv is depricated 
**This function was DEPRECATED in PHP 7.1.0, and REMOVED in PHP 7.2.0.**
[http://php.net/manual/en/function.mcrypt-create-iv.php](url)

you can use random_bytes() as an alternative.
[http://php.net/manual/en/function.random-bytes.php](url)

**follow these steps to resolve error.**

1. find file security.php location should be like this  (bonfire.dev\bonfire\ci3\core\Security.php)
2. find get_random_bytes() method
3. find `mcrypt_create_iv($length, MCRYPT_DEV_URANDOM)` and replace with  `random_bytes($length)`
4. that's it :+1:

**Comment by kalyan2k** [comment]
Thank you @truelineinfotech

**Comment by locvfx** [comment]
Thank you @truelineinfotech
